fleet_vehicle_category/models/fleet_vehicle.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2020 Stefano Consolaro (Ass. PNLUG - Gruppo Odoo <http://odoo.pnlug.it>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class VehicleCategory(models.Model):
+    """
+    A classification of vehicle based on functionality
+    eg. tow truck, ambulance, trailer, boat
+    """
+
+    _name = 'fleet.vehicle.category'
+    _description = 'Classification of vehicle, eg. tow truck, ambulance, trailer, boat'
+
+    # fields
+    # name
+    name = fields.Char('Category', 
+                       help='Eg. Tow truck, Ambulance, Trailer, Boat',
+                       )
+
+
+class FleetVehicle(models.Model):
+    """
+    Add field for category management to vehicle
+    """
+
+    _inherit = 'fleet.vehicle'
+
+    # category of vehicle
+    vehicle_category_id = fields.Many2one(
+        'fleet.vehicle.category',
+        'Vehicle Category',
+        help='Eg. Tow truck, Ambulance, Trailer, Boat',
+        )
```

Comment:
For me no, because if you set the category on the model you will be limited in the management because it can have only a category.
For example, if you buy two identical van and configure one for electrical maintenance an the other for idraulic maintenance: they will have two different categories but the same model.